crates/agglayer-aggregator-notifier/src/packer/mod.rs

```diff
@@ -146,76 +151,91 @@ where
 
         let state_store = self.state_store.clone();
         let config = self.config.clone();
+
         // Call the Provider
         let fut = Box::pin(
             async move {
-                let _tx = contract_call
-                    .send()
-                    .await
-                    .inspect(|tx| info!(hash, "Inspect settle transaction: {:?}", tx))
-                    // .map_err(SettlementError::ContractError)?
-                    .map_err(|e| {
-                        let error_str =
-                            RollupManagerRpc::decode_contract_revert(&e).unwrap_or(e.to_string());
+                // Send the transaction
+                let pending_tx = contract_call.send().await;
+                // wait for the receipt
+                let receipt = handle_pending_tx::<RollupManagerRpc>(
+                    pending_tx,
+                    hash.clone(),
+                    certificate_id,
+                    config,
+                )
+                .await?
+                // If the result is `None`, it means the transaction is no longer
+                // in the mempool.
+                .ok_or(Error::SettlementError {
+                    certificate_id,
+                    error: "No receipt hash returned, transaction still in mempool".to_string(),
+                })?;
 
+                match receipt.status {
+                    Some(n) if n == U64::zero() => {
+                        warn!(
+                            hash,
+                            "The transaction failed to settle the certificate {}", certificate_id
+                        );
+                        Err(Error::SettlementError {
+                            certificate_id,
+                            error: "SettlementTransaction failed".to_string(),
+                        })
+                    }
+                    None => {
                         error!(
-                        error_code = %e,
-                        error = error_str,
-                        hash,
-                            "Failed to settle the certificate {certificate_id}: {}", error_str);
+                            hash,
+                            "The transaction failed to settle the certificate {}", certificate_id
+                        );
 
-                        Error::SettlementError {
+                        Err(Error::SettlementError {
                             certificate_id,
-                            error: error_str,
+                            error: "SettlementTransaction failed due to no receipt status"
+                                .to_string(),
+                        })
+                    }
+                    Some(_) => {
+                        if let Err(error) = state_store.update_certificate_header_status(
+                            &certificate_id,
+                            &CertificateStatus::Settled,
+                        ) {
+                            error!(
```

Comment:
Here we just print error and continue. Is it enough?

Reply:
For now, we can't do much more, unfortunately.
Factorization are planned to address those